src/test/java/shop/service/BookServiceTest.java

```diff
@@ -0,0 +1,301 @@
+import shop.repository.spec.tools.book.spec.tools.impl.BookSpecificationBuilder;
+import shop.service.impl.BookServiceImpl;
+
+@ExtendWith(MockitoExtension.class)
+public class BookServiceTest {
+    private static final int PAGE_NUMBER = 0;
+    private static final int PAGE_SIZE = 20;
+    private static final Long ID_1L_CORRECT = 1L;
+    private static final Long INCORRECT_ID = -100L;
+    private static final String CATEGORY_NAME_CORRECT = "Test Category";
+    private static final String BOOK_TITLE_CORRECT = "Test Book";
+    private static final String BOOK_AUTHOR_CORRECT = "Test Author";
+    private static final String BOOK_ISBN_CORRECT = "Test Isbn";
+    private static final BigDecimal BOOK_PRICE_CORRECT = BigDecimal.TEN;
+
+    @InjectMocks
+    private BookServiceImpl bookService;
+    @Mock
+    private BookRepository bookRepository;
+    @Mock
+    private CategoryRepository categoryRepository;
+    @Mock
+    private BookMapper bookMapper;
+    @Mock
+    private BookSpecificationBuilder specificationBuilder;
+    private Pageable pageable;
+
+    @Test
+    @DisplayName("""
+            Verify createBook(...), return BookDto
+            """)
+    public void createBook_ValidCreateBookDto_returnBookDto() {
+        CreateBookRequestDto dto = new CreateBookRequestDto();
+        dto.setTitle(BOOK_TITLE_CORRECT);
+        dto.setAuthor(BOOK_AUTHOR_CORRECT);
+        dto.setIsbn(BOOK_ISBN_CORRECT);
+        dto.setPrice(BOOK_PRICE_CORRECT);
+        dto.setCategoryIds(new HashSet<>(Set.of(ID_1L_CORRECT)));
+
+        Category category = new Category();
+        category.setName(CATEGORY_NAME_CORRECT);
+
+        Book book = new Book();
+        book.setTitle(BOOK_TITLE_CORRECT);
+        book.setAuthor(BOOK_AUTHOR_CORRECT);
+        book.setIsbn(BOOK_ISBN_CORRECT);
+        book.setPrice(BOOK_PRICE_CORRECT);
+        book.setCategories(new HashSet<>(Set.of(category)));
+
+        BookDto expected = new BookDto();
+        expected.setId(ID_1L_CORRECT);
+        expected.setTitle(BOOK_TITLE_CORRECT);
+        expected.setAuthor(BOOK_AUTHOR_CORRECT);
+        expected.setIsbn(BOOK_ISBN_CORRECT);
+        expected.setPrice(BOOK_PRICE_CORRECT);
+        expected.setCategoryIds(new HashSet<>(Set.of(ID_1L_CORRECT)));
+
+        when(bookMapper.toBook(dto)).thenReturn(book);
+        when(categoryRepository.getReferenceById(ID_1L_CORRECT)).thenReturn(category);
+        when(bookRepository.save(book)).thenReturn(book);
+        when(bookMapper.toBookDto(book)).thenReturn(expected);
+
+        BookDto actual = bookService.createBook(dto);
+
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    @DisplayName("""
+            Verify getBookById(...), return BookDto
+            """)
+    public void getBookById_ValidId_returnBookDto() {
+        Category category = new Category();
+        category.setName(CATEGORY_NAME_CORRECT);
+
+        Book book = new Book();
+        book.setId(ID_1L_CORRECT);
```

Comment:
check if you have duplication and use separate methods for creating test data 
Also, you can create TestUtil class with methods that return test data and TestConstants for constants